archunit/src/main/java/com/tngtech/archunit/library/Architectures.java

```diff
@@ -294,6 +270,11 @@ public ArchRule because(String reason) {
             return ArchRule.Factory.withBecause(this, reason);
         }
 
+        @Override
+        public ArchRule allowEmptyShould(boolean allowEmptyShould) {
+            return withOptionalLayers(allowEmptyShould);
```

Comment:
Yes, it's a trade-off. But I think users won't use this method on a `LayeredArchitecture` exactly because `optionalLayers` makes more sense here. Thinking about it, we could write this in a Javadoc comment :thinking: